### PR TITLE
feat: Add max_quantity of ticket in order-statistics-event

### LINKED ISSUE #6710: Remove event-orga endpoint
It is not being used and redundant. Create proper sales stats API if needed

### PR DESCRIPTION
**Pull request by codedsun**
Fixes #6710 
feat: Add max_quantity of ticket in order-statistics-event

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

**Comment by codedsun** [comment]
@iamareebjamal - How do i include this resource in 
`http://127.0.0.1:5000/v1/users/8/events?page[size]=0`

through relationship?

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/api/order_statistics/events.py  1
         

Clones removed
==============
+ app/api/order_statistics/events.py  -4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4763932&bid=15882521)

File: app/api/order_statistics/events.py [hunk 1]
```diff
@@ -48,14 +49,17 @@ def tickets_count(self, obj):
...
         result = {
             'total': total or 0,
             'draft': draft or 0,
             'cancelled': cancelled or 0,
             'pending': pending or 0,
             'expired': expired or 0,
             'placed': placed or 0,
-            'completed': completed or 0
+            'completed': completed or 0,
+            'max_quantity': max_quantity or 0
```
Comment (iamareebjamal) [inline app/api/order_statistics/events.py]: Just max

File: app/api/schema/events.py [hunk 2]
```diff
@@ -272,6 +272,14 @@ def validate_timezone(self, data, original_data):
                                         related_view_kwargs={'event_id': '<id>'},
                                         schema='StripeAuthorizationSchema',
                                         type_='stripe-authorization')
+    order_statistics_event = Relationship(attribute='order_statistics_event',
```
Comment (codedsun) [inline app/api/schema/events.py]: @iamareebjamal - Please guide me here, this is not working when I hit with include. Am I doing it correct? or their is some another way

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- app/api/order_statistics/events.py  1
         

Clones removed
==============
+ app/api/order_statistics/events.py  -4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4763932&bid=15882521)

File: app/api/order_statistics/events.py [hunk 3]
```diff
@@ -1,6 +1,7 @@
 from flask_rest_jsonapi import ResourceDetail
 from marshmallow_jsonapi import fields
 from marshmallow_jsonapi.flask import Schema
+from marshmallow_jsonapi.flask import Relationship
```
Comment (niranjan94) [inline app/api/order_statistics/events.py]: Codacy Issue found: [Unused Relationship imported from marshmallow_jsonapi.flask](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4763932)

Comment (niranjan94) [inline app/api/order_statistics/events.py]: Codacy Issue found: ['marshmallow_jsonapi.flask.Relationship' imported but unused](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4763932)

**Comment by codedsun** [comment]
@iamareebjamal - I had left a comment. please help. Thanks

File: app/api/schema/events.py (hunk 2 above)
Comment (iamareebjamal) [reply app/api/schema/events.py]: See how it is done for other stuff. It is related to the urls, related and self. See the flask rest jsonapi docs. It's not obvious how to make it work where the data is unrelated to foreign key of the item. There is a guide in the docs on how to do it. I haven't done in previously. And docs are not great and very few people use the library and it is kinda abandoned

**Comment by iamareebjamal** [comment]
Answered

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/api/order_statistics/events.py  1
         

Clones removed
==============
+ app/api/order_statistics/events.py  -5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4763932&bid=15882521)

**Comment by codedsun** [comment]
@iamareebjamal - I have succesffully created the relationship but still I am getting error
for hitting url 
`http://127.0.0.1:5000/v1/events/38/relationships/order_statistics_event`

It says
```
{
    "errors": [
        {
            "status": 500,
            "source": "",
            "title": "Relation not found",
            "detail": "Event has no attribute order_statistics_event"
        }
    ],
    "jsonapi": {
        "version": "1.0"
    }
}
```

Hence I am again looking more to it, and will help this finalise asap

**Comment by iamareebjamal** [comment]
Relationship already exists - https://api.eventyay.com/v1/events/6/order-statistics

Don't know why you're working on that. You need to work on order statistics not working with include statement

**Comment by codedsun** [comment]
@iamareebjamal: No this is a seperate endpoint, the relationship doesn't exist in event schema. For the include to work, we need to have a relationship between event and orderstatistica which I did in this PR.

**Comment by iamareebjamal** [comment]
OK, so just add a property in event model `order_statistics` which returns the dict to be serialized and you're done

**Comment by codedsun** [comment]
>OK, so just add a property in event model order_statistics which returns the dict to be serialized and you're done

Then what about the relationships(order_statistics>event and vice versa), shall I remove them and only add this stated above?

**Comment by iamareebjamal** [comment]
> Then what about the relationships(order_statistics>event and vice versa), shall I remove them and only add this stated above?

No

> Also how will I call the function of OrderStatisticsEventSchema

No need to. Just return the dict

**Comment by codedsun** [comment]
@iamareebjamal - Tried the dict, but still missing something. the include is still not workng.

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/api/order_statistics/events.py  1
         

Clones removed
==============
+ app/api/order_statistics/events.py  -5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4763932&bid=15882521)

File: app/models/event.py [hunk 4]
```diff
@@ -447,6 +447,9 @@ def has_sessions(self):
     def has_speakers(self):
         return Speaker.query.filter_by(event_id=self.id).count() > 0
 
+    @property
+    def order_statistics(self):
+        return dict
```
Comment (iamareebjamal) [inline app/models/event.py]: Wow! What use is this for? You have to return the actual order statistics from here!

**Comment by iamareebjamal** [comment]
Of couse it won't. You haven't returned the order statistics in the attribute!

File: app/models/event.py (hunk 4 above)
Comment (codedsun) [reply app/models/event.py]: You mean to return `dict['order_statistics']`

**Comment by codedsun** [comment]
@iamareebjamal - I have not done this before and dont know how the returning of dict can make the order statistics available. Please guide me here, I am curious to learn for the future..

I have a doubt that are my relationships correct?

If possible please help me resolve this! 👍 Thanks

**Comment by iamareebjamal** [comment]
You have to return order statistics from the property. The actual order statistics. I don't know how to explain it more clearly. event.sessions returns list of sessions of an event. Hence, include? works with sessions. Similarly, event.order_statistics should return the order statistics of the event. How can I simplify more than that?

**Comment by codedsun** [comment]
@iamareebjamal - You are clear and have stated clearly but I don't know how do I do this(returning the order_statistics in event model), that's why I am asking, should I do like this? 
```
 @property
    def order_statistics(self):
        return dict('order_statistics')
```

I dont' know how to return that order_statistics without calling them from class
Reference:
> Also how will I call the function of OrderStatisticsEventSchema
No need to. Just return the dict

**Comment by iamareebjamal** [comment]
> I dont' know how to return that order_statistics without calling them from class

Who said you have to do it without calling them from class?

How is this order_statictics? - `dict('order_statistics')` This is not even valid python

**Comment by iamareebjamal** [comment]
> Also how will I call the function of OrderStatisticsEventSchema

Move the function to event model?

**Comment by codedsun** [comment]
@iamareebjamal - I misunderstood something might be https://github.com/fossasia/open-event-server/pull/6723#issuecomment-572449581

No problem, please excuse my mistake and tell me how should I do it as it's the last step reqd.

From OrderStatisticsEventSchema class, I shall call the functions seperately and return the dict from the property of order_statistics in event model? Is this to be done.

**Comment by iamareebjamal** [comment]
Movce the function to event model, no need to call schema from the event model

**Comment by codedsun** [comment]
> Movce the function to event model, no need to call schema from the event model

then how will this api (`http://127.0.0.1:5000/v1/events/38/order-statistics`) work, as they also use the same functions

Also, what was the point of relationship then we just made?

**Comment by iamareebjamal** [comment]
You'll return obj.order_statistics from the order statistics schema

**Comment by codedsun** [comment]
@iamareebjamal - I just added the sales count function for testing to work with include

The include order_statistics shows just the event id and not the sales-count

Response:
```  "included": [
        {
            "type": "order-statistics-event",
            "relationships": {
                "event": {
                    "links": {
                        "self": "/v1/events/37/relationships/event",
                        "related": "/v1/events/37"
                    }
                }
            },
            "id": "37",
            "links": {
                "self": "/v1/events/37/order-statistics"
            }
        },
```

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/api/order_statistics/events.py  1
         

Clones added
============
- app/models/event.py  3
         

Clones removed
==============
+ app/api/order_statistics/events.py  -3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4763932&bid=15882521)

File: app/api/schema/events.py [hunk 5]
```diff
@@ -272,6 +272,13 @@ def validate_timezone(self, data, original_data):
                                         related_view_kwargs={'event_id': '<id>'},
                                         schema='StripeAuthorizationSchema',
                                         type_='stripe-authorization')
+    order_statistics = Relationship(attribute='order_statistics',
+                                    self_view='v1.event_order_statistics',
+                                    self_view_kwargs={'id': '<id>'},
+                                    related_view_kwargs={'id': '<id>'},
+                                    related_view='v1.order_statistics_event_detail',
+                                    schema='OrderStatisticsEventSchema',
```
Comment (iamareebjamal) [inline app/api/schema/events.py]: Because nothing except id is present in OrderStatisticsEventSchema

Comment (codedsun) [reply app/api/schema/events.py]: @iamareebjamal You mean the fields present their in OrderStatisticsEventSchema
```
tickets = fields.Method("tickets_count")
    orders = fields.Method("orders_count")
    sales = fields.Method("sales_count")
```

these are not fetched, so how do i fetch them @iamareebjamal

**Comment by codedsun** [comment]
Closing as opened a new at #6735

**Comment by kushthedude** [comment]
@codedsun You don't need to close a PR if you mess up the comment history.
You can do your changes on a different branch and force push that branch to the branch of your PR.
Say on local you were developing at branch alpha and pushing them to alpha on Github and making a PR from it. Now you mess up locally on alpha. Instead, you just make fresh changes on beta locally and force push it on alpha on GitHub using git push -f origin beta:alpha 
Closing a PR every time some changes regarding commits are requested is bad practice -- It is a bad practice because all the previous discussions which took place in the original PR are lost.

**Pull request closed without merging**